### PR TITLE
Query and update Animals table

### PR DESCRIPTION
**Pull request by ridabensalem**
in this milestone of the vet-clinic project we have added the following requirements : 

- Add a column species of type string to your animals table
- Insert new data into the animal's table 
- Update the animal's table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to Pokemon for all animals that don't have species already set
- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction
- Write queries to answer the following questions:
1. How many animals are there?
2. How many animals have never tried to escape?
3. What is the average weight of animals?
4. Who escapes the most, neutered or not neutered animals?
5. What is the minimum and maximum weight of each type of animal?
6. What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -15,3 +15,34 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 -- Query 8: Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+/*Vet clinic database: query and update animals table*/
+-- Query 1: update the animals table by setting the species column to unspecified
+UPDATE animals SET species = 'unspecified';
+-- Query 2: Update the animals table by setting the species column to digimon for all animals that have a name ending in mon
```
Comment (Davidosky007) [inline queries.sql:21]: * As required in the [project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Inside%20a%20transaction%20update%20the%20animals%20table%20by%20setting%20the%20species%20column%20to%20unspecified.%20Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went%20back%20to%20the%20state%20before%20the%20transaction.) you are to 

> update the animals table by setting the species column to unspecified

> Verify that the change was made on the animal table 

> Then roll back the change and verify that the species columns went back to the state before the transaction. Kindly work on all these 🙏

File: queries.sql [hunk 2]
```diff
@@ -15,3 +15,34 @@ SELECT * FROM animals WHERE neutered = true;
...
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+-- Query 3: Update the animals table by setting the species column to pokemon for  all animals that don't have species already set.
+UPDATE animals SET species = 'pokemon' WHERE species='';
```
Comment (Davidosky007) [inline queries.sql]: * As stated in the requirement 

> verify that the change was made after updating animals' table 

> verify that change persists after commit, Kindly work on these 🙏

File: queries.sql [hunk 3]
```diff
@@ -15,3 +15,34 @@ SELECT * FROM animals WHERE neutered = true;
...
+-- Query 4:  Inside a transaction delete all records in the animals table then rollback
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (Davidosky007) [inline queries.sql]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Take%20a%20screenshot%20of%20the,the%20results%20of%20your%20actions.) 

> you are to delete all records in the animals table, then roll back the transaction.
> After the rollback verify if all records in the animals table still exists. below is an example 👇  
```
   BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ridabensalem 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Database transactions used 👍 
* Data modified and deleted using SQL 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ridabensalem** [comment]
Thank you for the great feedback all the changes have been made :smile:

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @ridabensalem , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All changes requested :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**